### ISSUE TITLE
Visual Studio 2017 RC Solution Rebuild Breaks References

### ISSUE THREAD

**Issue by cjbhaines** [fsprojects/Paket#2086]
### Description

Paket version: 4.0.0-alpha034

Issue symptoms: 
After running a dotnet restore, everything builds fine. Clicking "Rebuild Solution" under the Build menu clears all references and a dotnet restore must be run again.

### Repro Steps

1) Dowload https://dl.dropboxusercontent.com/u/49591225/Code/PaketTest_BuildDotnetCore.zip
2) Unzip solution
3) Run dotnet restore
4) Inside visual studio click Build -> Rebuild Solution

### Expected Behaviour

References are still loaded and solution builds fine

### Actual Behaviour

All references are unloaded and build fails

/cc @theimowski, @forki

**Comment by forki** [comment]
@davidKean @enricosada is this expected?

**Comment by enricosada** [comment]
@forki dunno, that is the same for proj without c# too?
What clean remove? files inside obj? paket create these on restore?

**Comment by cjbhaines** [comment]
Hey guys, just tried this on the new VS2017 release and still facing the same issues. Clean or rebuild requires a manual dotnet restore to be able to build again.

**Comment by forki** [comment]
@davkean do you know something about this?

**Comment by davkean** [comment]
Can you summarize how you are hooking up to restore? It sounds like you cleaning up files that you shouldn't on clean. Restore (yet) participate in Clean.

**Comment by forki** [comment]
@davkean 

This is the project file

    <Project ToolsVersion="15.0" Sdk="Microsoft.NET.Sdk.Web">
        <PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>netcoreapp1.1</TargetFramework>
        <PreserveCompilationContext>true</PreserveCompilationContext>
      </PropertyGroup>
      <ItemGroup>
        <Folder Include="wwwroot\" />
      </ItemGroup>
      <ItemGroup>
        <Content Update=".dockerignore">
          <DependentUpon>Dockerfile</DependentUpon>
        </Content>
        <Content Update="docker-compose.override.yml">
          <DependentUpon>docker-compose.yml</DependentUpon>
        </Content>
        <Content Update="docker-compose.vs.debug.yml">
          <DependentUpon>docker-compose.yml</DependentUpon>
        </Content>
        <Content Update="docker-compose.vs.release.yml">
          <DependentUpon>docker-compose.yml</DependentUpon>
        </Content>
      </ItemGroup>
      <Import Project="..\.paket\Paket.Restore.targets" />
    </Project>

this is the targets file:

    <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
      <PropertyGroup>
        
        <IsPaketRestoreTargetsFileLoaded>true</IsPaketRestoreTargetsFileLoaded>    
      </PropertyGroup>
      
      <UsingTask TaskName="Paket.Build.Tasks.PaketRestoreTask" AssemblyFile="PaketRestoreTask.dll" />
    
      <Target Name="PaketRestore" BeforeTargets="_GenerateRestoreGraphWalkPerFramework" >
        <Exec Command="$(MSBuildThisFileDirectory)\paket.exe restore --project $(MSBuildProjectFullPath)" />
    
         
        <PaketRestoreTask
          ProjectUniqueName="$(MSBuildProjectFullPath)"
          PackageReferences="@(PackageReference)"
          TargetFrameworks="$(TargetFramework)">
          <Output TaskParameter="NewPackageReferences" ItemName="PackageReference" />
          <Output TaskParameter="AlternativeConfigFile" ItemName="RestoreConfigFile" />        
        </PaketRestoreTask>
       </Target>
    
       <Target Name="FixPackRef" AfterTargets="_WalkEachTargetPerFramework" Returns="@(_PackageReferences)">
          
          <PaketRestoreTask
            ProjectUniqueName="$(MSBuildProjectFullPath)"
            PackageReferences="@(PackageReference)"
            TargetFrameworks="$(TargetFramework)">
            <Output TaskParameter="NewPackageReferences" ItemName="_PaketPackageReference" />
            <Output TaskParameter="AlternativeConfigFile" ItemName="RestoreConfigFile2" />
          </PaketRestoreTask>
          <ItemGroup>
            <_PackageReferences Include="%(_PaketPackageReference.Identity)">
              <TargetFramework>$(TargetFramework)</TargetFramework>
              <IncludeAssets></IncludeAssets>
              <ExcludeAssets></ExcludeAssets>
              <PrivateAssets></PrivateAssets>
              <Version>%(_PaketPackageReference.Version)</Version>
            </_PackageReferences>
          </ItemGroup>
          
        </Target>
    
    </Project>

Putting a PackageReference node directly into the csproj works for VS - then it will restore that file but not the packages that we have drawn via paket.

I also diffed the folders after "clean" - nothing changed. It's just that VS is not using our MSBuild task

**Comment by forki** [comment]
I believe @srivatsn's comment at https://github.com/dotnet/roslyn-project-system/issues/653#issuecomment-273697971 suggest we should do it differently. But I'm not really uszre I understand how.

**Comment by davkean** [comment]
CollectPackageReferences is the public extension point, whereas _GenerateRestoreGraphWalkPerFramework is not. Refactoring/bug fixes might result in _GenerateRestoreGraphWalkPerFramework being renamed/removed, in which case you will break.

What does `packet.exe restore` do vs `PacketRestoreTask`?

**Comment by forki** [comment]
paket.exe restore does the real work and downloads stuff.

PaketRestoreTask.dll just reads a file and (which was created by "paket.exe restore") to get the referenced packages into memory

**Comment by davkean** [comment]
I cannot get the project to build from the start, it appears none of the paket provided packages can be found:

1. Download solution
2. From a VS command-prompt, run `dotnet restore PacketTest.sln`
3. 'dotnet build PacketTest.sln'


```
Controllers\ValuesController.cs(5,17): error CS0234: The type or namespace name 'AspNetCore' does not exist in the na
mespace 'Microsoft' (are you missing an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Program.cs(6,17): error CS0234: The type or namespace name 'AspNetCore' does not exist in the namespace 'Microsoft' (
are you missing an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Program.cs(7,17): error CS0234: The type or namespace name 'AspNetCore' does not exist in the namespace 'Microsoft' (
are you missing an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(5,17): error CS0234: The type or namespace name 'AspNetCore' does not exist in the namespace 'Microsoft' (
are you missing an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(6,17): error CS0234: The type or namespace name 'AspNetCore' does not exist in the namespace 'Microsoft' (
are you missing an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(7,17): error CS0234: The type or namespace name 'Extensions' does not exist in the namespace 'Microsoft' (
are you missing an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(8,17): error CS0234: The type or namespace name 'Extensions' does not exist in the namespace 'Microsoft' (
are you missing an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(9,17): error CS0234: The type or namespace name 'Extensions' does not exist in the namespace 'Microsoft' (
are you missing an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(25,16): error CS0246: The type or namespace name 'IConfigurationRoot' could not be found (are you missing
a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(10,37): error CS0246: The type or namespace name 'Controller' could not be found (are
 you missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(9,6): error CS0246: The type or namespace name 'RouteAttribute' could not be found (a
re you missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(9,6): error CS0246: The type or namespace name 'Route' could not be found (are you mi
ssing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(13,10): error CS0246: The type or namespace name 'HttpGetAttribute' could not be foun
d (are you missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(13,10): error CS0246: The type or namespace name 'HttpGet' could not be found (are yo
u missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(20,10): error CS0246: The type or namespace name 'HttpGetAttribute' could not be foun
d (are you missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(20,10): error CS0246: The type or namespace name 'HttpGet' could not be found (are yo
u missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(27,10): error CS0246: The type or namespace name 'HttpPostAttribute' could not be fou
nd (are you missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(27,10): error CS0246: The type or namespace name 'HttpPost' could not be found (are y
ou missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(28,27): error CS0246: The type or namespace name 'FromBodyAttribute' could not be fou
nd (are you missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(28,27): error CS0246: The type or namespace name 'FromBody' could not be found (are y
ou missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(33,10): error CS0246: The type or namespace name 'HttpPutAttribute' could not be foun
d (are you missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(33,10): error CS0246: The type or namespace name 'HttpPut' could not be found (are yo
u missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(34,34): error CS0246: The type or namespace name 'FromBodyAttribute' could not be fou
nd (are you missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(34,34): error CS0246: The type or namespace name 'FromBody' could not be found (are y
ou missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(39,10): error CS0246: The type or namespace name 'HttpDeleteAttribute' could not be f
ound (are you missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Controllers\ValuesController.cs(39,10): error CS0246: The type or namespace name 'HttpDelete' could not be found (are
 you missing a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(28,39): error CS0246: The type or namespace name 'IServiceCollection' could not be found (are you missing
a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(35,31): error CS0246: The type or namespace name 'IApplicationBuilder' could not be found (are you missing
 a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(35,56): error CS0246: The type or namespace name 'IHostingEnvironment' could not be found (are you missing
 a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(35,81): error CS0246: The type or namespace name 'ILoggerFactory' could not be found (are you missing a us
ing directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
  Startup.cs(15,24): error CS0246: The type or namespace name 'IHostingEnvironment' could not be found (are you missing
 a using directive or an assembly reference?) [C:\Temp\ConvertTest\ConvertTest.csproj]
```

This is on the VS 2017 RTM.

**Comment by davkean** [comment]
Note the `<PackageReference>` provided packages, in this case, 'Microsoft.NETCore.App 1.1.1' can be found via NuGet.

**Comment by forki** [comment]
"dotnet restore PacketTest.sln" what's that?

usually we just go to the root of the repo and call "dotnet restore" - it worked for me,

**Comment by davkean** [comment]
They are equivalent.

**Comment by forki** [comment]
Mhm so maybe we use different dotnet sdks. Seems yours is not calling paket or something like that.  Anyways, how should I change the targets file? What do you recommend to do?

**Comment by davkean** [comment]
Okay, I found that I needed to run `dotnet restore` twice before paket would kick in, it's late here so I don't have time to investigate that - but you should treat that as a bug on your part that paket restore doesn't work first time you run it.

I know what's going on here - when we "Clean", we automatically kick off restore inside VS. I hinted at this problem a while ago on throughout emails. VS Restore isn't calling your target, everything works fine if you restore from command-line, but as soon as restore runs inside VS - it doesn't run your target.

Unfortunately, CollectPackageReferences was supposed to be the extension point for both VS and command-line restore for something like Paket kicking it - but it looks like we don't call it via command-line. I'll file that bug on our side.

As a workaround, changing the above to:

``` XML
<Target Name="PaketRestore" BeforeTargets="_GenerateRestoreGraphWalkPerFramework;CollectPackageReferences" >
```

Caused packet restore to run inside VS, but then your task failed to run due to you depending on a dll higher than what ships in the framework:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB4018	The "PaketRestoreTask" task failed unexpectedly.
System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
File name: 'System.Runtime, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Paket.Build.Tasks.PaketRestoreTask.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
	ConvertTest	C:\Temp\.paket\Paket.Restore.targets	13	
```

You either need to ship that dll with paket, or reduce your dependencies so that you don't depend on it. VS uses MSBuild 15 that runs on .NET Framework 4.6, so any tasks you right need to run on that version.

**Comment by forki** [comment]
ah cool. @davkean thanks so much for the help!

*  I changed the line in the targets file
* I downgraded the task to netstandard 1.3
* I released Paket 4 rc6

And this fixed VS 2017 build workflow for me. Cmd line workflow is working as well.

@cjbhaines So in order to get it working please update paket to rc6 and run "paket install" once.

> I needed to run dotnet restore twice

@davkean weird. Never seen this issue. Will keep an eye open.

**Comment by forki** [comment]
@davkean there is still one issue:

If you open VS it doesn't restore. you need to run "dotnet restore" at least once from cmd line. Then VS picks it up and everythings works as expected.

**Comment by cjbhaines** [comment]
Thanks for looking into this! Hopefully this enables us to start looking at a move to VS2017 and back to Paket. We are having to use project.json at the moment which is driving me mad!

Just hit a snag so blocked on testing this by #2182 - will test this when we get through that bug.

**Comment by forki** [comment]
just to clarify: #2182 is fixed, but the issue that VS2017 is not restoring after solution load / build is still open.

**Comment by cjbhaines** [comment]
I can confirm the rebuild/clean solution is fixed. @forki shall we create a new bug for the not restoring after solution load issue?

**Comment by forki** [comment]
No. I reopened

Am 14.03.2017 17:05 schrieb "Chris Haines" <notifications@github.com>:

**Comment by forki** [comment]
@davkean I created a new zip which paket rc. [VSLoad.zip](https://github.com/fsprojects/Paket/files/843632/VSLoad.zip)

The following gif shows the problem:

![vs](https://cloud.githubusercontent.com/assets/57396/23937356/7990eb0c-0957-11e7-8df9-85b99e1e34f1.gif)

Basically if you open VS nothing works and a build from within Vs doesn't trigger the restore.
But if you manually restore from outside then magically things work.

**Comment by davkean** [comment]
I turned on design-time build logging, and first of all, packet.exe took > 30 seconds to complete, then PackageRestoreTask failed with:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB4018	The "PaketRestoreTask" task failed unexpectedly.
System.IO.FileNotFoundException: Could not load file or assembly 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
File name: 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Paket.Build.Tasks.PaketRestoreTask.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
	ConvertTest	C:\Temp\VSTests\.paket\Paket.Restore.targets	21	
```

This eventually shows up in the error list after packet.exe finished.

**Comment by forki** [comment]
how can I turn it on?

**Comment by davkean** [comment]
Here's the build log: https://gist.github.com/davkean/90f140c6b6906e2af0c887d53f5e8d46

I'm updating [our docs](https://github.com/dotnet/roslyn-project-system/blob/master/docs/design-time-builds.md) right now  how to turn this on with the new installer.

**Comment by forki** [comment]
the error is a bit weird. I mean I used .netstandard1.3 I thought that's what's available inside VS

**Comment by davkean** [comment]
Any "System.*" that ends up in your build directory needs to be installed next to your task.

**Comment by forki** [comment]
after dotnet build I have:

![image](https://cloud.githubusercontent.com/assets/57396/23945570/4412ea50-0977-11e7-99da-6d723e624742.png)

**Comment by forki** [comment]
^^ so I don't see what I can do

**Comment by davkean** [comment]
Yeah, hold on let me finish something first.

**Comment by davkean** [comment]
I've updated our docs to show how to turn on design-time build logging: https://github.com/dotnet/roslyn-project-system/blob/master/docs/design-time-builds.md#visual-studio-2017.

Yeah, my mistake I forgot MSBuild is the actual host. This is harder than I thought, as you're basically going to need to have different dependencies depending on the host CLR - I'm going to have to ask @dsplaisted, @rainersigwald and @ericstj for ideas here. This is a lot harder than it should be to write a task that works across both CoreCLR MSBuild and desktop MSBuild.

**Comment by forki** [comment]
@davkean Ok cool I followed your docs and I have:

![image](https://cloud.githubusercontent.com/assets/57396/23947166/d73cf7fc-097d-11e7-9f90-cf647cc848b5.png)

So that registry hack works. thx

**Comment by forki** [comment]
using System.IO;
    using Microsoft.Build.Framework;
    using Microsoft.Build.Utilities;
    
    namespace Paket.Build.Tasks
    {
        public class PaketRestoreTask : Task
        {
    
            /// <summary>
            /// Full path to the msbuild project.
            /// </summary>
            [Required]
            public string ProjectUniqueName { get; set; }
    
            [Required]
            public ITaskItem[] PackageReferences { get; set; }
    
            /// <summary>
            /// Target frameworks to apply this for. If empty this applies to all.
            /// </summary>
            public string TargetFrameworks { get; set; }
    
            /// <summary>
            /// Output items
            /// </summary>
            [Output]
            public ITaskItem[] NewPackageReferences { get; set; }
    
            [Output]
            public string AlternativeConfigFile { get; set; }
    
            public override bool Execute()
            {            
                var fileInfo = new FileInfo(ProjectUniqueName);
                var fileName = Path.Combine(fileInfo.Directory.FullName, "obj", fileInfo.Name + ".references");
                var lines = System.IO.File.ReadAllLines(fileName);
                var list = new System.Collections.Generic.List<ITaskItem>();
                list.AddRange(PackageReferences);
                char[] delimiterChars = {  ','};
    
                foreach (var line in lines)
                {
                    var splitted = line.Split(delimiterChars);
                    if(splitted.Length < 2)
                        break;
                    ITaskItem dependency = new TaskItem(splitted[0].Trim());
                    dependency.SetMetadata("Version", splitted[1].Trim());
                    list.Add(dependency);
                }
                
                this.NewPackageReferences = list.ToArray();
    
                AlternativeConfigFile = Path.Combine(fileInfo.Directory.FullName, "obj", fileInfo.Name + ".NuGet.Config");
    
                return true;
            }
        }
    }


^^ is my task (pretty trivial). Maybe it's possible to inline that into the Paket.Restore.targets file!?

**Comment by forki** [comment]
mhm that's also more complicated than expected since dotnet restore doesn't like the hello world task from https://msdn.microsoft.com/en-us/library/dd722601.aspx

**Comment by enricosada** [comment]
@forki no inline for netcore msbuild. Build the dll and include in nupkg

**Comment by forki** [comment]
that's what I did. but it's not working in VS as @davkean showed

**Comment by forki** [comment]
@enricosada found a way to do everything in the targets file: https://github.com/fsprojects/Paket/commit/49e691f050ca5b388ad835d31f46cc2e7d838a83#diff-e8d5cbc6647c87d83063161d4a423537

**Comment by davkean** [comment]
You might want to look at the 30 seconds delay that I noticed paket had during a design-time build - that's not an acceptable wait for a design-time. These builds are supposed to be extremely quick  - this will block IntelliSense from being updated and other services in the project system. Design-time gets kicked off very often for every change to the project system, including simply by dropping a new file on disk. On top of this, this will affect other projects in the solution even if they are not using paket, as this will block available nodes that we can build on.

**Comment by forki** [comment]
That is paket downloading packages on very first start. Nuget would have to
do that as well, right?

Am 16.03.2017 6:40 vorm. schrieb "David Kean" <notifications@github.com>:

**Comment by davkean** [comment]
We don't block design-time builds on restore, that happens in the background. You can see progress in the status bar - that's NuGet downloading packages.

**Comment by forki** [comment]
Ok so our Integration is not enough.

Ideally we detect that we are in design time and call paket with special
parameter so that it doesn't download.

In second step (from different design time build target?) we call paket
again and let it download?


Am 16.03.2017 7:05 vorm. schrieb "David Kean" <notifications@github.com>:

**Comment by forki** [comment]
@davkean is that what you do with nuget? How would that look like? Thx for the help

**Comment by davkean** [comment]
NuGet inside VS doesn't run using MSBuild (hence why restore was different between command-line and VS). They have a service that runs in the background and kicks off restore when the project changes.

I need to think about how this should work - I suspect, we should long term get NuGet to call an MSBuild target to do the restore - that's the point at which you would kick in. I need to chew on this to think about how this should work in the short-term.

**Comment by forki** [comment]
Cool. Thank you!

Am 16.03.2017 08:45 schrieb "David Kean" <notifications@github.com>:

NuGet inside VS doesn't run using MSBuild (hence why restore was different
between command-line and VS). They have a service that runs in the
background and kicks off restore when the project changes.

I need to think about how this should work - I suspect, we should long term
get NuGet to call an MSBuild target to do the restore - that's the point at
which you would kick in. I need to chew on this to think about how this
should work in the short-term.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
<https://github.com/fsprojects/Paket/issues/2086#issuecomment-286981086>,
or mute the thread
